src/renderer/src/components/song/song-detail/SongDetail.tsx

```diff
@@ -52,8 +67,14 @@ const ProgressBar = () => {
       onValueCommit={handleSeekEnd}
       animate
     >
-      <Slider.Track class="flex h-7 items-center rounded-xl bg-thick-material p-1">
-        <Slider.Range class="block h-5 rounded-l-lg bg-surface" />
+      <Slider.Track
+        class="flex h-7 items-center rounded-xl p-1"
+        style={{ "background-color": props.averageColor }}
+      >
+        <Slider.Range
+          class="block h-5 rounded-l-lg"
+          style={{ "background-color": props.averageColor }}
```

Comment:
The style prop is not working on the `<Styles.Range/>` component, to make it work you'll need to create the following function on `src/renderer/src/lib/css.utils.ts`:

``` ts
type Style = string | JSX.CSSProperties | undefined;
export function sn(...styles: Style[]): JSX.CSSProperties {
  return styles.reduce<JSX.CSSProperties>((acc, style) => {
    if (typeof style !== "object") {
      return acc;
    }

    return {
      ...acc,
      ...style,
    };
  }, {});
}
```

After that you can update the `<Slider.Range />` (on `src/renderer/src/components/slider/SliderRange.tsx`) to use it.

``` tsx
import { useSlider } from "./Slider";
import { sn } from "@renderer/lib/css.utils";
import { Component, JSX } from "solid-js";

const SliderRange: Component<JSX.IntrinsicElements["span"]> = (props) => {
  const state = useSlider();

  return (
    <span
      {...props}
      style={sn(
        state.transitionStyleValue(),
        {
          width: `${state.percentage()}%`,
          "pointer-events": "none",
          "transition-property": "width",
        },
        props.style,
      )}
    />
  );
};

export default SliderRange;
```